cmd/podman/shared/pod.go

```diff
@@ -134,4 +137,129 @@ func CreatePortBindings(ports []string) ([]ocicni.PortMapping, error) {
 	return portBindings, nil
 }
 
+func GetPodsWithFilters(r *libpod.Runtime, c *cliconfig.PodPsValues) ([]*libpod.Pod, error) {
+
+	var pods []*libpod.Pod
+	var err error
+	if c.Latest {
+		pod, err := r.GetLatestPod()
+		if err != nil {
+			return nil, err
+		}
+		pods = append(pods, pod)
+	} else {
+		pods, err = GenerateFilterFunctionAndGetPods(r, strings.Split(c.Filter, ","))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return pods, nil
+}
+
+func GenerateFilterFunctionAndGetPods(r *libpod.Runtime, filters []string) ([]*libpod.Pod, error) {
```

Comment:
Function comment would be appreciated

Reply:
Also, should this be public? Looks internal-only

Reply:
fixed. made private.